compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -117,22 +116,10 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
         else paramTypes
       tp.derivedLambdaType(paramNames, paramTypes1, resultType1)
     case tp: PolyType =>
-      tp.derivedLambdaType(tp.paramNames, tp.paramInfos, elimRepeated(tp.resultType))
+      tp.derivedLambdaType(tp.paramNames, tp.paramInfos, elimRepeated(tp.resultType, isJava))
     case tp =>
       tp
 
-  def transformTypeOfTree(tree: Tree)(using Context): Tree =
```

Comment:
Is there a rationale (except tests pass) that we can remove these? I mean, it's great that we can, but I wonder why the overrides were there in the first place.

Reply:
I think that the TimeTravellingTreeCopier (which didn't exist back when ElimRepeated was first added) takes care of any necessary type update in tree.